game-core/src/main/java/games/strategy/triplea/delegate/AAInMoveUtil.java

```diff
@@ -255,12 +253,11 @@ private static PlayerID findDefender(final Collection<Unit> defendingUnits, fina
         && defendingUnits.stream().anyMatch(Matches.unitIsOwnedBy(territory.getOwner()))) {
       return territory.getOwner();
     }
-    for (final Unit u : defendingUnits) {
-      if (u != null && u.getOwner() != null) {
-        return u.getOwner();
-      }
-    }
-    return PlayerID.NULL_PLAYERID;
+    return defendingUnits.stream()
+        .filter(u -> u != null && u.getOwner() != null)
+        .findFirst()
+        .map(Unit::getOwner)
```

Comment:
Ditto previous comment

Reply:
appreciate the point, but the previous comment did not compile; before diving into this for by-hand updates, I'd rather fix some root causes for the null units; fixing streaming syntax to be nicer is misguided IMO.